xds/src/main/java/io/grpc/xds/XdsServerBuilder.java

```diff
@@ -67,37 +79,36 @@ public XdsServerBuilder xdsServingStatusListener(
     return this;
   }
 
-  /**
-   * Unsupported call. Users should only use {@link #forPort(int, ServerCredentials)}.
-   */
-  @DoNotCall("Unsupported. Use forPort(int, ServerCredentials) instead")
-  public static ServerBuilder<?> forPort(int port) {
-    throw new UnsupportedOperationException(
-        "Unsupported call - use forPort(int, ServerCredentials)");
+  public static XdsServerBuilder forPort(int port) {
+    return forPort(port, InsecureServerCredentials.create());
   }
 
   /** Creates a gRPC server builder for the given port. */
   public static XdsServerBuilder forPort(int port, ServerCredentials serverCredentials) {
-    NettyServerBuilder nettyDelegate = NettyServerBuilder.forPort(port, serverCredentials);
+    checkNotNull(serverCredentials, "serverCredentials");
+    InternalProtocolNegotiator.ServerFactory originalNegotiatorFactory =
+            InternalNettyServerCredentials.toNegotiator(serverCredentials);
+    ServerCredentials wrappedCredentials = InternalNettyServerCredentials.create(
+            new FilterChainMatchingNegotiatorServerFactory(originalNegotiatorFactory));
+    NettyServerBuilder nettyDelegate = NettyServerBuilder.forPort(port, wrappedCredentials);
     return new XdsServerBuilder(nettyDelegate, port);
   }
 
   @Override
   public Server build() {
-    return buildServer(new XdsClientWrapperForServerSds(port));
+    checkState(isServerBuilt.compareAndSet(false, true), "Server already built!");
+    AtomicReference<FilterChainSelector> filterChainSelectorRef = new AtomicReference<>();
+    InternalNettyServerBuilder.eagAttributes(delegate, Attributes.newBuilder()
+            .set(ATTR_FILTER_CHAIN_SELECTOR_REF, filterChainSelectorRef)
+            .build());
```

Comment:
Is it possible to move lines 100-103 into the XdsServerWrapper constructor? I don't see any reason to do this here unless the intent is to not call Netty code from inside `XdsServerWrapper`

Reply:
> because this attributes communicates between XdsServerWrapper and transport layer.

Another way of saying you don't want to pollute `XdsServerWrapper` with any transport layer logic which is fair although I don't know when we'll actually have non-Netty XdsServers. Otherwise it is best to move these 4 lines into the `XdsServerWrapper` constructor and hide the logic there.